adapters/InMobi/InMobiAdapter/InMobiMediatedNativeAppInstallAd.m

```diff
@@ -0,0 +1,233 @@
+                self.mappedImages = @[ [[GADNativeAdImage alloc] initWithURL:imageURL scale:[[imageDictionary objectForKey:ASPECT_RATIO] floatValue]] ];
+                self.mappedIcon = [[GADNativeAdImage alloc]
+                                   initWithURL:iconURL
+                                   scale:[[iconDictionary objectForKey:ASPECT_RATIO] floatValue]];
+                if([self respondsToSelector:inmobiMediatedNativeAppInstallAdSuccessful]){
+                    [self inmobiMediatedNativeAppInstallAdSuccessful:self];
+                }
+            }
+            else{
+                dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+                    UIImage *img, *icon;
+                    [imageStringURL stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+                    
+                    UIImage *cachedImage = [imageCache objectForKey:imageStringURL];
+                    if(cachedImage){
+                        img = cachedImage;
+                    }
+                    else
+                    {
+                        NSData *imageData = [NSData dataWithContentsOfURL:[NSURL URLWithString:imageStringURL]];
+                        img = [UIImage imageWithData:imageData];
+                        if(img != nil){
+                            [imageCache setObject:img forKey:imageStringURL];
+                        }
+                    }
+                    if(img != nil){
+                        self.mappedImages = @[ [[GADNativeAdImage alloc] initWithImage:img] ];
+                    }
+                    else{
+                        [self inmobiMediatedNativeAppInstallAdFailed];
+                    }
+                
+                    [iconStringURL stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+                    
+                    UIImage *cacheIcon = [imageCache objectForKey:iconStringURL];
+                    if(cacheIcon){
+                        icon = cacheIcon;
+                    }
+                    else{
+                        NSData *iconData = [NSData dataWithContentsOfURL:[NSURL URLWithString:iconStringURL]];
+                        icon = [UIImage imageWithData:iconData];
+                        if(icon != nil){
+                            [imageCache setObject:icon forKey:iconStringURL];
+                        }
+                    }
+                    
+                    if(icon != nil){
+                        self.mappedIcon = [[GADNativeAdImage alloc] initWithImage:icon];
+                    }
+                    else{
+                        [self inmobiMediatedNativeAppInstallAdFailed];
+                    }
+                
+                    [self inmobiMediatedNativeAppInstallAdSuccessful:self];
+                });
+            }
+        }
+    }
+    return self;
+}
+
+- (NSString *)headline {
+    if([[self.nativeAdContentDictionary objectForKey:TITLE] length]){
```

Comment:
This check needs to be performed during the initialization of this class and a failure callback needs to be sent if any of the required assets (Headline, Image, Body, App Icon and Call to Action for [app install ads](https://support.google.com/admob/answer/6240809)) are missing.

Same for all the required assets below.